functions/src/modules/TE.functions.ts

```diff
@@ -37,45 +35,48 @@ export const processAllotment = functions.https.onCall(
       .once('value');
     if (!userRef.exists()) {
       throw new functions.https.HttpsError(
-        'not-found',
-        "Assignee wasn't found!."
+        'invalid-argument',
+        "Assignee wasn't found!"
       );
     }
 
     // Update the task
-    const allTasks = [];
-    tasks.forEach(async taskId => {
-      const regex = /(\w+)-(\d+)-(\d+)/g;
-      const taskIdMatch = regex.exec(taskId);
-      const list = taskIdMatch[1];
+    const allTasks = await Promise.all(
+      tasks.map(async (taskId: string) => {
+        const list = helpers.extractListFromFilename(taskId);
 
-      await db.ref(`/edited/${list}/${taskId}/trackEditing`).update({
-        status: 'Given',
-        assignee: assignee,
-        givenTimestamp: moment().format('MM/DD/YYYY'),
-      });
+        const teDbRef = db.ref(`/edited/${list}/${taskId}/trackEditing`);
 
-      // Getting the tasks list to be used when notifying the assignee (Step 3)
-      const taskRef = await db
-        .ref(`/edited/${list}/${taskId}/trackEditing`)
-        .once('value');
-      allTasks.push(taskRef.val());
-    });
+        await teDbRef.update({
+          status: 'Given',
+          assignee: assignee,
+          givenTimestamp: moment().format(),
+        });
+
+        // Getting the tasks list to be used when notifying the assignee
+        const taskRef = await teDbRef.once('value');
```

Comment:
The result of `once` [is called a snapshot](https://firebase.google.com/docs/reference/admin/node/admin.database.Reference#once), so it's better not to introduce a confusion by calling this `taskRef`.